src/Identity/Endpoints/src/IdentityBearerAuthenticationHandler.cs

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity.Endpoints.DTO;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Identity.Endpoints;
+
+internal sealed class IdentityBearerAuthenticationHandler : SignInAuthenticationHandler<IdentityBearerOptions>
+{
+    private const string BearerTokenPurpose = $"Microsoft.AspNetCore.Identity.Endpoints.IdentityBearerAuthenticationHandler:v1:BearerToken";
+
+    private static readonly AuthenticateResult TokenMissing = AuthenticateResult.Fail("Token missing");
+    private static readonly AuthenticateResult FailedUnprotectingToken = AuthenticateResult.Fail("Unprotected token failed");
+    private static readonly AuthenticateResult TokenExpired = AuthenticateResult.Fail("Token expired");
+
+    private readonly IDataProtectionProvider _fallbackDataProtectionProvider;
+
+    public IdentityBearerAuthenticationHandler(
+        IOptionsMonitor<IdentityBearerOptions> optionsMonitor,
+        ILoggerFactory loggerFactory,
+        UrlEncoder urlEncoder,
+        ISystemClock clock,
+        IDataProtectionProvider dataProtectionProvider)
+        : base(optionsMonitor, loggerFactory, urlEncoder, clock)
+    {
+        _fallbackDataProtectionProvider = dataProtectionProvider;
+    }
+
+    private IDataProtectionProvider DataProtectionProvider
+        => Options.DataProtectionProvider ?? _fallbackDataProtectionProvider;
+
+    private ISecureDataFormat<AuthenticationTicket> BearerTokenProtector
+        => Options.BearerTokenProtector ?? new TicketDataFormat(DataProtectionProvider.CreateProtector(BearerTokenPurpose));
+
+    protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
+    {
+        // If there's no bearer token, forward to cookie auth.
+        if (await GetBearerTokenOrNullAsync() is not string token)
+        {
+            return Options.MissingBearerTokenFallbackScheme is string fallbackScheme
+                ? await Context.AuthenticateAsync(fallbackScheme)
+                : TokenMissing;
+        }
+
+        var ticket = BearerTokenProtector.Unprotect(token);
+
+        if (ticket?.Properties?.ExpiresUtc is null)
+        {
+            return FailedUnprotectingToken;
+        }
+
+        if (Clock.UtcNow >= ticket.Properties.ExpiresUtc)
```

Comment:
Does this account for clock skew?

Reply:
No. And I don't see how it really can and work on multiple different machines using the same data protector in load balanced scenarios. This is the same way the `CookieAuthenticationHandler` handles expiration.